### PR TITLE
Change dataTaskWithHTTPMethod visibility to public

### PR DESCRIPTION
**Pull request by smartinspereira**
Change -dataTaskWithHTTPMethod visibility from private to public.
#2379, #2380, #2404, #2862, #3361

### CODE REVIEW

File: AFNetworking/AFHTTPSessionManager.h [hunk 1]
```diff
@@ -299,6 +299,27 @@ NS_ASSUME_NONNULL_BEGIN
...
+ @param failure A block object to be executed when the task finishes unsuccessfully, or that finishes successfully, but encountered an error while parsing the response data. This block has no return value and takes a two arguments: the data task and the error describing the network or parsing error that occurred.
+
+ @see -dataTaskWithRequest:uploadProgress:downloadProgress:completionHandler:
+ */
+- (NSURLSessionDataTask *)dataTaskWithHTTPMethod:(NSString *)method
+                                       URLString:(NSString *)URLString
+                                      parameters:(nullable id)parameters
+                                  uploadProgress:(nullable void (^)(NSProgress *uploadProgress)) uploadProgress
+                                downloadProgress:(nullable void (^)(NSProgress *downloadProgress)) downloadProgress
+                                         success:(void (^)(NSURLSessionDataTask *, id))success
```
Comment (jshier) [inline AFNetworking/AFHTTPSessionManager.h]: The block arguments need to be updated with the proper nullability.

Comment (smartinspereira) [reply AFNetworking/AFHTTPSessionManager.h]: Looks good. Let's merge this?

**Comment by ElfSundae** [comment]
Missing `headers` parameter
https://github.com/AFNetworking/AFNetworking/blob/a1f208b79e1338310e27108b8dc55c7343b656b1/AFNetworking/AFHTTPSessionManager.m#L338-L346

**Comment by jshier** [comment]
This is fixed in my `afnetworking4` branch and will go out with the release.

**Pull request merged**